cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
Oh I think you misunderstood, the kid `public` and `private` are unique - like a fixed primary key. However, if multiple keys exists, they have a different id, for example `1234` or `public-1234`. This way we only have one public kid in the set. Does that make sense?